shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -692,11 +702,25 @@ void Engine::WarmupSkps(fml::BasicTaskRunner* concurrent_task_runner,
   // tell concurrent task runner to deserialize all skps available from
   // the asset manager
   concurrent_task_runner->PostTask([raster_task_runner, skp_warmup_surface,
-                                    &surface_producer]() {
+                                    &surface_producer, skp_names,
+                                    completion_callback]() {
     TRACE_DURATION("flutter", "DeserializeSkps");
-    std::vector<std::unique_ptr<fml::Mapping>> skp_mappings =
-        flutter::PersistentCache::GetCacheForProcess()
-            ->GetSkpsFromAssetManager();
+    std::vector<std::unique_ptr<fml::Mapping>> skp_mappings;
+    if (skp_names) {
+      auto asset_manager =
```

Comment:
Can you inject the asset manager here as a parameter to WarmupSkps?

Reply:
sure